api/core/v1beta2/cluster_types.go

```diff
@@ -722,10 +825,10 @@ type MachineDeploymentTopology struct {
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
 
-	// machineHealthCheck allows to enable, disable and override
+	// healthCheck allows to enable, disable and override
 	// the MachineHealthCheck configuration in the ClusterClass for this MachineDeployment.
```

Comment:
Wondering if this should be

```suggestion
         // healthCheck allows to enable, disable and override control plane HealthCheck settings from the ClusterClass.
```